src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -213,6 +213,96 @@ internal partial class CodeFixService : ForegroundThreadAffinitizedObject, ICode
             return result.ToImmutableAndFree();
         }
 
+        public async Task<ImmutableArray<CodeFixCollection>> GetFixesAsync(Document document, string diagnosticId, TextSpan range, CancellationToken cancellationToken)
+        {
+            var diagnostics = (await _diagnosticService.GetDiagnosticsForSpanAsync(document, range, diagnosticId, cancellationToken: cancellationToken).ConfigureAwait(false)).ToList();
```

Comment:
this is all duplicate code I just cooked up for demo. you probably want some of refactoring to remove copy and paste code.

Reply:
updated